libs/core/src/lib/fixed-card-layout/fixed-card-layout.component.ts

```diff
@@ -0,0 +1,157 @@
+import { CdkDragDrop, CdkDropList, moveItemInArray } from '@angular/cdk/drag-drop';
+import {
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ContentChildren,
+    Directive,
+    ElementRef,
+    EventEmitter,
+    HostListener,
+    Output,
+    QueryList,
+    TemplateRef,
+    ViewChild
+} from '@angular/core';
+
+const CARD_MINIMUM_WIDTH = 320; // in px; 20rem max card size
+const CARD_GAP_WIDTH = 16; // gap=1rem==16px
+
+@Directive({ selector: '[fdCardDef]' })
+export class CardDefinitionDirective {
+    constructor(public template: TemplateRef<any>) {}
+}
+
+class Layout {
+    constructor(public noOfColumns: number, public screenSize: number) {}
+}
+
+class CardDropped {
+    constructor(public container: CdkDropList, public previousIndex: number, public currentIndex: number) {}
+}
+
+type CardColumn = CardDefinitionDirective[];
+
+@Component({
+    selector: 'fd-fixed-card-layout',
+    templateUrl: './fixed-card-layout.component.html',
+    styleUrls: ['./fixed-card-layout.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class FixedCardLayoutComponent implements AfterViewInit {
+    /** @hidden */
+    @ContentChildren(CardDefinitionDirective)
+    cards: QueryList<CardDefinitionDirective>;
```

Comment:
Probably we need to listen for cads changes, sice user can add it dynamically (using ngIf or ngFor for cards rendering)

Reply:
fixed.